src/test/sigencoding_tests.cpp

```diff
@@ -21,11 +21,17 @@ static valtype SignatureWithHashType(valtype vchSig, SigHashType sigHash) {
 static void CheckSignatureEncodingWithSigHashType(const valtype &vchSig,
                                                   uint32_t flags) {
     ScriptError err = ScriptError::OK;
-    BOOST_CHECK(CheckDataSignatureEncoding(vchSig, flags, &err));
 
     const bool hasForkId = (flags & SCRIPT_ENABLE_SIGHASH_FORKID) != 0;
+    // isTaproot requires Schnorr sigs and BIP341 sighash
+    const bool isTaproot = (flags & SCRIPT_REQUIRE_TAPROOT_SIGNATURES) != 0;
     const bool is64 = (vchSig.size() == 64);
 
+    // isTaproot is never true for OP_CHECKDATASIG, but we test it anyway to
```

Comment:
Can you elaborate on this?

Reply:
Hmm, so we use the `SCRIPT_REQUIRE_TAPROOT_SIGNATURES` script flag to enforce BIP341 and Schnorr signatures for the key spend path, which is only ever set for that sig check (using `EvalChecksig`).

Maybe changing it to `isTRKeySpend` helps? It shouldn't be too long otherwise the predicates become unreadable.

I could also change the flag to `SCRIPT_TAPROOT_KEY_SPEND_PATH`, do you think that would be more clear?